app/controllers/workshops_controller.rb

```diff
@@ -11,6 +11,10 @@ def show
   def rsvp
     redirect_to :back, notice: t('workshops.registration_not_open') unless @workshop.invitable_yet?
 
+    unless @workshop.date_and_time.future?
+      return redirect_to :back, notice: t('messages.already_taken_place')
```

Comment:
```suggestion
      return redirect_to :back, notice: t('messages.already_taken_place')  unless @workshop.date_and_time.future?
```

Reply:
@simbabque Maybe a better way around this that is also easier to read (and should resolve the codeclimate issue) is to extend the workshop model to perform the check about the workshop being available for rsvp, and calling that through here. I think adding a more specific message is a nice to have, but not essential as I _think_ the refreshed page might already have some text indicating that the event has already taken place.


e.g.

```ruby
def available_for_rsvp?
   invitable_yet? && date_and_time.future?
end